resolver/dns/dns_resolver_test.go

```diff
@@ -34,6 +34,11 @@ import (
 )
 
 func TestMain(m *testing.M) {
+	// Set a very small duration for the resolver rate limit to avoid tests
+	// running for long time.
+	dc := replaceDNSResRate(1 * time.Second)
```

Comment:
Done. Had to add a field in `dnsResolver` and read it from there in `watcher()` instead of from the global var to make the race detector happy.

Reply:
This doesn't sound right - you may be changing the var while the resolver is running?  If you set it before building the resolver, and reset it after closing the resolver, I would think it would be OK.

Reply:
Done. You are right. Thanks.